### ISSUE TITLE
How does this differ from Scribus Generator?

### ISSUE THREAD

**Issue by luzpaz** [bjarnesvanberg/Scribus-Data-Merger#1]
https://github.com/berteh/ScribusGenerator

**Comment by bjarnesvanberg** [comment]
When I tried to use ScribusGenerator to merge CSV data with a Scribus file I found that ScribusGenerator was generating a new PDF file as output for each row of input data. What I want this project to do is to add the data-merged content right there in the Scribus file. You can then edit the Scribus file further and export to PDF when you are ready to do so.

**Comment by luzpaz** [comment]
Interesting. What OSs have you tested it on?  
CC @aoloe @berteh

**Comment by bjarnesvanberg** [comment]
My project or ScribusGenerator? In either case I have only been running them on Windows 7. My project is in a working state now so you can see the differences yourselves. 

Now that you mention this, it just dawned on me that I have only tested with the Scribus 1.5.0. It would in fact probably not work with Scribus 1.4.x.

From Help -> About Scribus:
Scribus Version 1.5.0svn
19 Maj 2015
Build ID: C-_-T-_-C1.12.18-Windows-64bit

**Comment by luzpaz** [comment]
@bjarnesvanberg thanks for the follow-up. 
The questions was meant for Scribus-Data-Merge. 

Another question: would it make sense to merge the 2 projects, Scribus-Data-Merge & ScribusGenerator?

CC @aoloe @berteh

**Comment by bjarnesvanberg** [comment]
I can see why that would make sense when looking at the functionality. However, ScribusGenerator is working on the XML structure of a Scribus file and generating a new temporary file with the variables replaced with data for each PDF file created. It then exports the temporary file as a PDF. Scribus Data Merger on the other hand is utilizing the Scripter API to manipulate the current document.

When looking at the functionality I think there is a justified case for each of those scripts as ScribusGenerator is better suited for repetition of multiple pages which Scribus Data Merger is not meant to handle at the moment. Scribus Data Merger will insert new pages into the existing Scribus document and then allowing the user to make more edits to the document before exporting. Usability-wise I think it will be hard to explain to the user how the two ways of merging differs while keeping the user interface simple. 

When looking at the code I think there is a clear overlap w.r.t. the data we are collecting from the user and the way the replacement code is handled. The issue about how each page object is accessed and modified differs quiet a lot (XML vs Scripter API) which could be a challenge to merge. 

I think it's a nice proposal but in order to make it happen I need input on how to go about the issues I have talked about here. 

CC @aoloe @berteh

**Comment by berteh** [comment]
Hello @bjarnesvanberg.

Nice project of yours. Thanks @luzpaz for the notification.
Merging would be an option (or even simply working together on a single approach... as long as all important functionalities are covered ;)

What I like about this project more is that it could maybe go a bit beyond repeating a single page: could you think of repeating only a selected portion of the whole document for each row?  I don't know how to identify it, but it would be great! You could populate an array in the text body... or generate a deck of cards (eg 2 pages recto + verso for each row). Quite flexible!

What I like about my project more is that it works with scribus 1.4... and is fast as lightning to generate the slas (not the pdfs though). I was simply planning to add the feature to merge all generated slas in a single one (at the xml level)... but as that would make it quite a duplicate of this project I'll freeze that until we decide together on the best way to go further.

One option to me is to have a quick chat about it, if you'd like. Or we can just keep on writing our ideas & preferences here ;)

PS: just a side-note for this Data-Merge: I've been asked to support other csv delimiters, and think it would make sense here too.

**Comment by bjarnesvanberg** [comment]
Yes lets have a chat about it at some point. I have updated my Github profile with a link to my Google plus profile. I guess we can hook up via Google hangouts that way :) (Though I might not be available before Monday)

**Comment by berteh** [comment]
a gHangout somewhere next week seems perfect... and yes indeed, this project does unfortunately not work in scribus 1.4.x as the internal scripter API was not developed enough before 1.5.

**Comment by berteh** [comment]
@luzpaz how important is it to keep supporting 1.4.x for such a feature? is 1.5 scheduled to go public any time soon?

**Comment by aoloe** [comment]
@berteh, 1.6 is unlikely to be released anytime soon... but every five minutes that are put in the mainteannce/development of 1.4 are five minutes lost for 1.6... this is mostly true for the few resources the two and a half programmers in the team that are currently active.
so: i would only target 1.6 (1.5 being the development branch that will lead to the stable 1.6)

**Comment by berteh** [comment]
bjarnesvanberg and me had a good chat about it and decided to bring the nice Data-Merge functionalities into Generator (ie output to single sla), because the later is much faster and flexible, and because we had no time enough to expose all needed elements in the Scribus API to make it work the other way round.  We'll try to work together to make the "new joint" Generator even better ;)

**Comment by luzpaz** [comment]
@berteh that is exciting! Glad you and @bjarnesvanberg are collaborating. 
Just wanted to CC @jainbasil to bring to his attention all the attention scripter is getting.
